Legal Problem! Revert before it's too late!

thermatk opened the pull request:
Hello FastHub Developers!
I was a contributor here before and back then we didn't grant our copyright to @k0shk0sh, sign a CLA or anything, so everyone contributed under the GPL producing respective GPL code(the cancerous nature of GPL as it is).
Recently @k0shk0sh made a mistake and changed the license file(with changes that are void according to GPL anyway) without asking explicit permission to do so from all contributors to the repo. 
The problem should be easily fixable by reverting this misstep, although if there were any new contributors who have sent code in the timeframe from 3rd September till now their code is licensed under the new text... which has void terms, so it's a much lesser problem.
Great that we got it before it's too late and people started suing each other

k0shk0sh commented:
Well, as far as I can tell, your whole code was taken down therefore you have nothing really here unless you prove me wrong and please if there is any, let me know and i'll take it down immediately because I don't want any of your code in my base. 

and for the rest of the contributors, they are basically fine with that and I can ensure you that because they are helping out open source with knowing that there are in app purcahses within the app.

Now come the part where you wanna play smart, i'm still with you taking out the pro features, i have asked nicely in https://github.com/thermatk/FastHub-Libre/issues/6 and again will ask here, should I proceed with my PR?

stek29 commented:
@k0shk0sh 
> and for the rest of the contributors, they are basically fine with that and I can ensure you that because they are helping out open source with knowing that there are in app purcahses within the app.

This is not how it works. You can't do illegal stuff and say it's legal because nobody told you it was.
It's still illegal to hack into someone's wifi for example, even if WiFi owner doesn't notice it and says nothing.

k0shk0sh commented:
@stek29 what is illegal of me preventing any other fork of using PRO features to sell it or provide it for free? I didn't prevent no one from forking the repo or use its code, but I wrote something to the open source community that should respect as I respect theirs. 

so I'm really unsure what is the problem in here.

stek29 commented:
@k0shk0sh 
> what is illegal

Violating GPL.

thermatk commented:
@k0shk0sh so it was an "unanimous decision" like the last time?
You should have 55(ok, maybe less, after you find all the code that is destroyed and replaced in whole, etc, take into account your employees and things like that) EXPLICIT CONSENTS.

It's the GPL world, so it's not really your code anymore. You can take out YOUR code, riddled with contribution holes, and release a new version closing the source from this moment, but everything ever done before remains GPL. Everything forked and modified from it should be released in source form, if you distribute binaries, with the binaries, as GPL requires, and so on

thermatk commented:
@k0shk0sh once again. 
What you want could be done, but the app is under GPL, so the pro features should've been added to a separate app, as the easiest way to do that. Otherwise everything you add to the binary of this app should be added to the source and... the source can be used in any way those who have it would like, because of the so-called "cancerous" nature of GPL

k0shk0sh commented:
And you right, considering your last pull it was at 2.x which has no pro features only later they came in. 

If you still wanna talk about that decision well, your code didn't disappear I made a branch out of it before it even gone and was gonna be your branch before you started all of this. Anyway the past is past.

I still want pro features to be removed from your fork.

stek29 commented:
@k0shk0sh 
The issue is that you've licensed **everything** under GPL, and now you're saying "WAIT I CHANGED MY MIND NO PAY ME NO GPL 4 U".
That's not how it works.

You would have to change license, or do di
You have multiple options:
- Introduce CLA, wait for all contributors to sign it. When they do, you can finally change license.
- License your code under different terms, but keep GPL for parts of code coming from contributors.
- Revert all commits not made by you, and re-do them. But then it'd mean you're using GPL code and if you change license it may be still illegal (It's like getting sources of linux, then saying "hey it's my code" and licensing it under different license)

Even if you do this, that would not affect your old commits.
License change does not have retroactive force.

Now, let's imagine you've changed your license. You'd have to stop using any GPL code, continuous integration and other third party services also might require you to license code under free opensource license and not "k0shk0shGPL" or whatever you name it.

stek29 commented:
@k0shk0sh 
> I still want pro features to be removed from your fork.

Then at least make it easy.
Feel free to *want* pro features to be removed. But you can't *force* and you don't even have legal right to *demand* them to be removed.

thermatk commented:
@k0shk0sh 
this PR is a friendly move, nothing else. Once again, it's not about me or my code, it's about ALL people who ever contributed.
In my first comment I've asked about CLA being introduced, but it probably wasn't, so you have no copyright over all changes except YOURS. So you can't change license, dual-license or whatever, without explicitly asking for consents of EVERYONE holding copyrights. 
Expecting something about their attitude when they contributed and that they have the same view over GPL is not the same as seeing that they have sent PRs to a repo with the GPLv3 license

k0shk0sh commented:
@stek29 I don't want to change the license and I do respect the license but what I can't get is that you guys arguing on something that we all understand and get, but you keep going away from my main request, PRO features belong to in app purchase and because of you I had to write such thing to the license, why abuse open source? Is this how you guys appreciate it? If so, please do what you like.

thermatk commented:
@k0shk0sh so remove the illegal clause from the file and stop abusing open source
Then we might talk, without threats

I will send you to see OsmAnd+, [which is a paid app in Google Play](https://play.google.com/store/apps/details?id=net.osmand.plus&hl=ru), but gets built in F-Droid and distributed for free, you might present your arguments and so on 👍

xenreon commented:
@k0shk0sh it technically does violate the gpl with the way you want so not much to argue there.

That said @thermatk care to remove the _liberated_ pro features from your fork? That's practically abusing open source and a slap to his face. We can continue this all we want but can't really legally stop you can we? Just like lucky patcher.

yakov116 commented:
@thermatk u are such a 2 year old.

Why don't you be a man and agree that you ARE MAD (that he removed your hard work) and that you really want to get him back.

The way you are handling this is just showing how stupid you can be in order to take revenge. 

One thing I can say, until now I looked up at how you allowed the whole thing to pass. Now I look at you as if you are someone that has no life. Thinks that by not providing their info to Google they will never get it (ha ha think again). You will turn your head upside down to make sure they don't have it. 

Grow up!

k0shk0sh commented:
You still taking things to serious ain't you? I didn't get the part that includes threatening and I shouldn't really ask for that, you should by yourself avoid doing such thing without my approval to either include pro features as free or whatsoever. 

I do upload in every release ton GitHub, I do have everything unlocked without any restriction if you build the source yourself and I have zero problems with that, but what I don't accept is that the number of people who bought pro features or the students who shared their ids with me via email could find out that they can get them for free from foss. Its even a bad image for foss. 

I know that I can't do anything about these pirated apps but I know for sure that devs should respect one another. 


I'll accept this PR to remove that sentence however I do expect the community pay respect in return.

Cheers Ruslan.

yakov116 commented:
@k0shk0sh I'm proud of you!

k0shk0sh commented:
@xenreon the app is for developers just like you and me, when I made everything transparent I had the image of every developer will be the same, but unfortunately looking at their reviews in the playstore changed this view and made realize that not all devs do respect other people's work nor the effort they put in. But that doesn't change my point of view of Ruslan, we talked for awhile and I guess we know ourselves attitude. That's why I came to him and asked him nicely but he thought I was threatening so he threaten me first 😅

thermatk commented:
Let me clarify my position

A discussion about my life and personality, that's not very cool. The word of the day is "respect", so please stay professional. I will ignore that, thank you @k0shk0sh 

It's really sad that you might see this as revenge or something.I'll tell you how I see it then. 

While I seriously laugh at phrases about abusing open source in the repo, which almost literally shattered my beliefs in open source, as @yakov116 correctly mentioned I got over it. This was the hardest hit in my contributors-hobby-career I ever got, with the least possible respect and "abusing open source" community as far as I can imagine. I have been a contributor(a proud collaborator!) here, I've been more engaged than in other repos, more than I initially planned("be friendly, bring another app I like to f-droid"), because I really loved the app. I've been sitting with my laptop at nights catching bugs in the notifications thingy and many other parts. However small that might seem, work is work, and I've not only got F-Droid to include the app, which wasn't as easy as usual, but also didn't sacrifice functionality for all kinds of users. One night, when I was half-offline, I get a notification, which also turned out to be accidental, that all I've done is removed. I came to ask, nicely back then, because I didn't understand that. Was there an issue with the code I felt proud to overlook? Maybe I missed a discussion on Slack? Nope. Later, the whole discussion with my comments(not really aggressive, not even close to current level of discussion) was also removed for some greater good of the community or to stop open source abuse and all we got was some Soviet-style release about a "unanimous decision" to drop f-droid because some other feature was hard to implement. You could've dropped a letter, you know.

I got over it by understanding that it's alright and that open source is not a friendly kindergarden, but a place where people can do whatever they are ok with personally and want to do, if it's allowed by the rules. Honestly, FastHub is the project I treat more formal than any other in this sense, can't deny that. So, I forked it to have my own project and now you come there, go through the source(copying me deleting outdated stuff, eh? 😸 ) and find something I never even advertised. I have been removing the endless googleplay checks and for fun made myself equal to what a play store user can do, very offensive. 

If the community would be less toxic here, you would be discussing how abusing for open source is having blacklists of users in the app, not my sovereign fork.
I will respect the GPL and ask you to do the same and I'm open to any discussions, but I also have a life throughout the day, so my responses may be a bit slow.
Have fun.

k0shk0sh commented:
Wow, such a dramatic response, full of drama and attention. 

Could we stop this right now? 


Again, dropping foss was one of the best decision I made to make FastHub as it is now, so no regret in that decision. 


So get over it and move on, you aren't welcome here so accept this. 


P.S: blacklisted users isn't working, but that was a good addition for your drama. 


Cheers.

k0shk0sh commented:
No more drama & comments.


This post is locked.


Thank you all.

k0shk0sh commented:
@yakov116 you shouldn't have said that and I asked you not to involve.


Could you please do not involve in any discussion that happen between me him? 


That's not cool, I got that you mad and dislike him, but do not start a war in my repo.
